lte/gateway/c/session_manager/UpfMsgManageHandler.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*****************************************************************************
+  Source      	UpfMsgManagerHandler.cpp
+  Version     	1.0
+  Date       	2020/01/17
+  Product     	SessionD
+  Subsystem   	UPF converged Landing object in SessionD
+  Author/Editor Venu Kumar Gurrapu
+  Description 	Acts as 5G Landing object in SessionD & start 5G related flow
+*****************************************************************************/
+#include <chrono>
+#include <thread>
+
+#include <google/protobuf/util/time_util.h>
+#include "UpfMsgManageHandler.h"
+#include "magma_logging.h"
+#include "GrpcMagmaUtils.h"
+
+using grpc::Status;
+
+namespace magma {
+/**
+ * SetInterfaceForUserPlaneHandler processes gRPC requests for the sessionD
+ * This composites the all the request that comes from UPF
+ */
+
+UpfMsgManageHandler::UpfMsgManageHandler(
+    std::shared_ptr<SessionStateEnforcer> enforcer, SessionStore& session_store)
+    : session_store_(session_store), conv_enforcer_(enforcer) {}
+
+/**
+ * Node level GRPC message received from UPF
+ * during startup
+ */
+
+void UpfMsgManageHandler::SetUPFNodeState(
+    ServerContext* context, const UPFNodeState* node_request,
+    std::function<void(Status, SmContextVoid)> response_callback) {
+  auto& request = *node_request;
+  // Print the message from UPF
+  PrintGrpcMessage(static_cast<const google::protobuf::Message&>(request));
+  MLOG(MDEBUG) << "Node UPF details :";

```

Comment:
not sure where this is going to be executed, but if this is an asynch function, shouldnt we add its execution inside a `runInEventBaseThread` to make sure we run in it in the base thread

Reply:
most of things are already in "runInEventBaseThread ", now I  changed most of the code.  Please re-review after my next commit/review.